tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/AbstractTychoMapping.java

```diff
@@ -261,20 +258,45 @@ protected Charset getPrimaryArtifactCharset() {
         return StandardCharsets.UTF_8;
     }
 
-    protected abstract void initModel(Model model, Reader artifactReader, File artifactFile)
-            throws ModelParseException, IOException;
+    protected abstract void initModel(Model model, Reader artifactReader, File artifactFile) throws IOException;
 
     protected static Properties getBuildProperties(File dir) throws IOException {
+        return loadProperties(new File(dir, "build.properties"));
+    }
+
+    static Properties loadProperties(File propertiesPath) throws IOException {
         Properties properties = new Properties();
-        File buildProperties = new File(dir, "build.properties");
-        if (buildProperties.exists()) {
-            try (FileInputStream stream = new FileInputStream(buildProperties)) {
+        if (propertiesPath.isFile()) {
+            try (FileInputStream stream = new FileInputStream(propertiesPath)) {
                 properties.load(stream);
             }
         }
         return properties;
     }
 
+    static Supplier<Properties> getPropertiesSupplier(File propertiesFile) {
+        Mutable<Properties> props = new MutableObject<>();
```

Comment:
Why do we need the Mutable stuff here? I think you can replace this with a simple inner class of a Supplier and use an instance field instead of wrapping the mutable in a lamda.

Reply:
> Why do we need the Mutable stuff here? I think you can replace this with a simple inner class of a Supplier and use an instance field instead of wrapping the mutable in a lamda.

Good suggestion. A anonymous inner class makes this even simpler.